RecoLocalTracker/SiPixelRecHits/plugins/alpaka/PixelCPEFastParamsESProducerAlpaka.cc

```diff
@@ -0,0 +1,121 @@
+#include <memory>
+#include <string>
+#include <alpaka/alpaka.hpp>
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ESProducer.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EventSetup.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ModuleFactory.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+#include "RecoLocalTracker/Records/interface/TkPixelCPERecord.h"
+#include "RecoLocalTracker/SiPixelRecHits/interface/alpaka/PixelCPEFastParamsCollection.h"
+#include "RecoLocalTracker/SiPixelRecHits/interface/PixelCPEFastParamsHost.h"
+
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "MagneticField/Engine/interface/MagneticField.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+#include "RecoLocalTracker/ClusterParameterEstimator/interface/PixelClusterParameterEstimator.h"
+
+#include "CondFormats/DataRecord/interface/SiPixelGenErrorDBObjectRcd.h"
+#include "RecoLocalTracker/Records/interface/PixelCPEFastParamsRecord.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+  template <typename TrackerTraits>
+  class PixelCPEFastParamsESProducerAlpaka : public ESProducer {
+  public:
+    PixelCPEFastParamsESProducerAlpaka(edm::ParameterSet const& iConfig);
+    std::unique_ptr<PixelCPEFastParamsHost<TrackerTraits>> produce(const PixelCPEFastParamsRecord& iRecord);
+
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  private:
+    edm::ESGetToken<MagneticField, IdealMagneticFieldRecord> magfieldToken_;
+    edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> pDDToken_;
+    edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> hTTToken_;
+    edm::ESGetToken<SiPixelLorentzAngle, SiPixelLorentzAngleRcd> lorentzAngleToken_;
+    edm::ESGetToken<SiPixelLorentzAngle, SiPixelLorentzAngleRcd> lorentzAngleWidthToken_;
+    edm::ESGetToken<SiPixelGenErrorDBObject, SiPixelGenErrorDBObjectRcd> genErrorDBObjectToken_;
+
+    edm::ParameterSet pset_;
+    bool useErrorsFromTemplates_;
+  };
+
+  using namespace edm;
+
+  template <typename TrackerTraits>
+  PixelCPEFastParamsESProducerAlpaka<TrackerTraits>::PixelCPEFastParamsESProducerAlpaka(const edm::ParameterSet& p)
+      : ESProducer(p), pset_(p) {
+    auto const& myname = p.getParameter<std::string>("ComponentName");
+    auto const& magname = p.getParameter<edm::ESInputTag>("MagneticFieldRecord");
+    useErrorsFromTemplates_ = p.getParameter<bool>("UseErrorsFromTemplates");
+
+    auto cc = setWhatProduced(this, myname);
```

Comment:
Strictly speaking the `ComponentName` parameter is redundant, because the framework already provides `appendToDataLabel` that would do all this for you (but ok, the pattern exists already).